firmware/application/apps/ble_app.cpp

```diff
@@ -0,0 +1,403 @@
+                       &check_log,
+#if USE_CONSOLE
+                       &console });
+#else
+                       &recent_entries_view,
+                       &recent_entry_detail_view });
+#endif
+
+    recent_entry_detail_view.hidden(true);
+
+    recent_entries_view.on_select = [this](const BleRecentEntry& entry) {
+        nav_.push<BleRecentEntryDetailView>(entry);
+    };
+
+    // field_frequency.set_value(get_freq_by_channel_number(37));
+    field_frequency.set_step(2000000);
+
+    check_log.set_value(logging);
+
+    check_log.on_select = [this](Checkbox&, bool v) {
+        str_log = "";
+        logging = v;
+    };
+
+    options_region.on_change = [this](size_t, int32_t i) {
+        if (i == 0) {
+            field_frequency.set_value(get_freq_by_channel_number(37));
+            channel_number = 37;
+        } else if (i == 1) {
+            field_frequency.set_value(get_freq_by_channel_number(38));
+            channel_number = 38;
+        } else if (i == 2) {
+            field_frequency.set_value(get_freq_by_channel_number(39));
+            channel_number = 39;
+        }
+
+        baseband::set_btle(channel_number);
+    };
+
+    options_region.set_selected_index(0, true);
+
+    logger = std::make_unique<BLELogger>();
+
+    if (logger)
+        logger->append(LOG_ROOT_DIR "/BLELOG_" + to_string_timestamp(rtc_time::now()) + ".TXT");
+
+    // Auto-configure modem for LCR RX (will be removed later)
+    baseband::set_btle(channel_number);
+
+    receiver_model.enable();
+}
+
+void BLERxView::on_data(BlePacketData* packet) {
+    std::string str_console = "";
+
+    if (!logging) {
+        str_log = "";
+    }
+
+    switch ((ADV_PDU_TYPE)packet->type) {
+        case ADV_IND:
+            str_console += "ADV_IND";
+            break;
+        case ADV_DIRECT_IND:
+            str_console += "ADV_DIRECT_IND";
+            break;
+        case ADV_NONCONN_IND:
+            str_console += "ADV_NONCONN_IND";
+            break;
+        case SCAN_REQ:
+            str_console += "SCAN_REQ";
+            break;
+        case SCAN_RSP:
+            str_console += "SCAN_RSP";
+            break;
+        case CONNECT_REQ:
+            str_console += "CONNECT_REQ";
+            break;
+        case ADV_SCAN_IND:
+            str_console += "ADV_SCAN_IND";
+            break;
+        case RESERVED0:
+        case RESERVED1:
+        case RESERVED2:
+        case RESERVED3:
+        case RESERVED4:
+        case RESERVED5:
+        case RESERVED6:
+        case RESERVED7:
+        case RESERVED8:
+            str_console += "RESERVED";
+            break;
+        default:
+            str_console += "UNKNOWN";
+            break;
+    }
+
+    // str_console += to_string_dec_uint(value);
+
+    str_console += " Len: ";
+    str_console += to_string_dec_uint(packet->size);
+
+    str_console += "\n";
+
+    str_console += "Mac";
+    str_console += ":" + to_string_hex(packet->macAddress[0], 2);
```

Comment:
I see two places where you're to_string on a MAC. Definitely consider adding a string_format helper at some point.

Reply:
Good idea, I'm deciding what information I want in the logs. I'll probably reformat this when I make the BLE logging more informative.